openshift/console
# ODC-7188: On importing application via import from git flow pac should be configured for the application

**Fixes**: 

https://issues.redhat.com/browse/ODC-7188

**Solution Description**: 

Added PAC Import Strategy to the Git Import Form. Users now can create Pipeline Repository directly from the Git Import Form if the Git Repository contains `.tekton` folder.

**ACs Completed:**

- [x] When user enters the repo url check for pipeline in .tekton directory of the repo
- [x] If a pipeline exists and a repository CR is already available for the repo show a message to the user and disable the create button
- [x] If a pipeline exists and a repository CR is not available for the repo update the import strategy
- [x] Add the configuration options of the Add git repo form to import from git form under the import strategy section
- [x] If a github app is already configured for the cluster show both `Use github app` and `Setup a webhook` options and if not then show only `Setup a webhook` option
- [x] Hide application field from general section
- [x] Hide resources section
- [x] Hide pipelines section
- [x] Hide advanced options
- [x] When user clicks on create, create repo CR and navigate the user to the repo details page
- [x] This feature should only be available when Pipelines operator is installed and PAC is enabled

**Screenshots / Gifs for design review**: 

**Operator is installed and GitHub App is set**

https://user-images.githubusercontent.com/47265560/205735735-008ee694-3383-4ad1-bf0e-bc762fb30dd2.mp4

**Demonstrating the change of strategies when Repository already exists**

https://user-images.githubusercontent.com/47265560/206153975-5ccbc8a2-9ea1-4683-90bf-b3d174a63159.mp4

**Screenshot of the Edit Strategy Piece**
![screencapture-localhost-9000-import-ns-openshift-pipelines-2022-12-16-19_28_14](https://user-images.githubusercontent.com/47265560/208114079-b2d5028b-96ae-4d6c-bb3d-9687a4691738.png)

**Screenshot of the PAC Strategy form when GitHub App is selected**
![screencapture-localhost-9000-import-ns-openshift-pipelines-2022-12-16-19_22_41](https://user-images.githubusercontent.com/47265560/208113580-ec8951d9-3118-4084-9e6e-a9c338db7079.png)

**Screenshot of the PAC Strategy form when Setup Webhook is selected**
![screencapture-localhost-9000-import-ns-openshift-pipelines-2022-12-16-19_23_09](https://user-images.githubusercontent.com/47265560/208113703-cac00c51-fb13-48af-a559-abd6bf8b530d.png)




**Unit test coverage report**: 

Fixed

**Test setup:**

0. Install the Pipelines Operator version above 1.8 (This comes with **PAC** pre-installed, which is a prerequisite)
1. Go to the Git Import Add Form.
2. Paste a Git URL having a `.tekton` folder. 
a. GitHub: https://github.com/Lucifergene/oc-pipe
b. GitLab: https://gitlab.com/avikkundu/oc-pipe
c: Bitbucket: https://bitbucket.org/avikkundu/oc-pipe
3. If GitHub App is set up, you will see the `Use GitHub app` and `Setup a webhook` options, else only the `Setup a webhook` Form.
4. Fill in the Git Secret and select a Webhook Secret. Optionally, edit the Name of the Repository.
5. If you have already created a repository with the URL you have entered as the Git URL, you will get an error stating to change your strategy. Click on the `Edit Strategy Link` to see the available strategies.


**Browser conformance**:
